rest-api/dao/participant_summary_dao.py

```diff
@@ -150,6 +150,53 @@ def _get_dna_isolates_sql_and_params():
       params
   )
 
+def _get_sample_status_time_sql_and_params():
+  """Gets SQL that to update enrollmentStatusCoreStoredSampleTime field
+  on the participant summary.
+  """
+
+  dns_test_list = config.getSettingList(config.DNA_SAMPLE_TEST_CODES)
+
+  status_time_sql = '%s' % ','.join(["""COALESCE(sample_status_%s_time, '3000-01-01')""" % item
+                                     for item in dns_test_list])
+
+  sub_sql = """
+  (
+    SELECT 
+      MAX(
+        consent_for_electronic_health_records_time,
+        questionnaire_on_the_basics_time,
```

Comment:
You may want to generate the SQL for these questionnaire fields based on config (similar to status_time_sql), like we do when running Python here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/questionnaire_response_dao.py#L41

That way when the config changes we won't have to update the code.